From dc6e3fe3b0f1683e9030003a691f069796ca6f5a Mon Sep 17 00:00:00 2001
From: Sachin Prabhu <sprabhu@redhat.com>
Date: Mon, 28 Jul 2014 12:15:55 -0400
Subject: [fs] cifs: Do not take a reference to the page in cifs_readpage_worker()

Message-id: <1406549822-10376-3-git-send-email-sprabhu@redhat.com>
Patchwork-id: 86746
O-Subject: [RHEL7.1 PATCH 02/69] [bz1082049]cifs: Do not take a reference to the page in cifs_readpage_worker()
Bugzilla: 1082049
RH-Acked-by: J. Bruce Fields <bfields@redhat.com>
RH-Acked-by: Steve Dickson <SteveD@redhat.com>

We do not need to take a reference to the pagecache in
cifs_readpage_worker() since the calling function will have already
taken one before passing the pointer to the page as an argument to the
function.

Signed-off-by: Sachin Prabhu <sprabhu@redhat.com>
Reviewed-by: Jeff Layton <jlayton@redhat.com>
Signed-off-by: Steve French <smfrench@gmail.com>
(cherry picked from commit a9e9b7bc15a32ec5b0679704e70f3ffeecfaadd8)

Signed-off-by: Sachin Prabhu <sprabhu@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/cifs/file.c b/fs/cifs/file.c
index 2821b42..39f33e6 100644
--- a/fs/cifs/file.c
+++ b/fs/cifs/file.c
@@ -3417,6 +3417,9 @@ static int cifs_readpages(struct file *file, struct address_space *mapping,
  return rc;
 }
 
+/*
+ * cifs_readpage_worker must be called with the page pinned
+ */
 static int cifs_readpage_worker(struct file *file, struct page *page,
  loff_t *poffset)
 {
@@ -3428,7 +3431,6 @@ static int cifs_readpage_worker(struct file *file, struct page *page,
  if (rc == 0)
   goto read_complete;
 
- page_cache_get(page);
  read_data = kmap(page);
  /* for reads over a certain size could initiate async read ahead */
 
@@ -3455,7 +3457,6 @@ static int cifs_readpage_worker(struct file *file, struct page *page,
 
 io_error:
  kunmap(page);
- page_cache_release(page);
 
 read_complete:
  return rc;
-- 
1.7.1